pkg/syncer/cnsoperator/util/util.go

```diff
@@ -107,33 +107,43 @@ func GetTKGVMIP(ctx context.Context, vmOperatorClient client.Client, dc dynamic.
 		return "", err
 	}
 
-	var networkName string
-	for _, networkInterface := range virtualMachineInstance.Spec.NetworkInterfaces {
+	var networkNames []string
+	for _, networkInterface := range virtualMachineInstance.Spec.Network.Interfaces {
 		// The assumption is that a TKG VM will have only a single network interface.
 		// This logic needs to be revisited when multiple network interface support
 		// is added.
-		networkName = networkInterface.NetworkName
+		networkNames = append(networkNames, networkInterface.Network.Name)
 	}
-	log.Debugf("VirtualMachine %s/%s is configured with network %s", vmNamespace, vmName, networkName)
+	log.Debugf("VirtualMachine %s/%s is configured with networks %v", vmNamespace, vmName, networkNames)
 
 	var ip string
 	var exists bool
 	if network_provider_type == NSXTNetworkProvider {
-		virtualNetworkInstance, err := dc.Resource(virtualNetworkGVR).Namespace(vmNamespace).Get(ctx,
-			networkName, metav1.GetOptions{})
-		if err != nil {
-			return "", err
+		for _, networkName := range networkNames {
+			virtualNetworkInstance, err := dc.Resource(virtualNetworkGVR).Namespace(vmNamespace).Get(ctx,
+				networkName, metav1.GetOptions{})
+			if err != nil {
+				return "", err
+			}
+			log.Debugf("Got VirtualNetwork instance %s/%s with annotations %v",
+				vmNamespace, virtualNetworkInstance.GetName(), virtualNetworkInstance.GetAnnotations())
+			ip, exists = virtualNetworkInstance.GetAnnotations()[snatIPAnnotation]
```

Comment:
Can you add a comment here:
// Pick the network interface which has the snatIPAnnotation